pkg/test/server.go

```diff
@@ -0,0 +1,156 @@
+package test
+
+import (
+	"encoding/csv"
+	"fmt"
+	"log"
+	"net"
+	"os"
+
+	"github.com/miekg/dns"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+var domainToIP = loadRecords("data/test-300.csv")
+
+const truncateSize = 512
+
+type Server struct {
+	Addr    string
+	TCPPort int
+	UDPPort int
+}
+
+func (s *Server) parseUDPAddr() string {
+	return fmt.Sprintf("%v:%v", s.Addr, s.UDPPort)
+}
+
+func (s *Server) parseTCPAddr() string {
+	return fmt.Sprintf("%v:%v", s.Addr, s.TCPPort)
+}
+
+func (s *Server) Run() {
+	g := new(errgroup.Group)
+	udpSrv := &dns.Server{Addr: s.parseUDPAddr(), Net: "udp", Handler: &udpHandler{}}
+	defer func() {
+		err := udpSrv.Shutdown()
+		if err != nil {
+			logrus.Errorf("shutting down UDP server: %v", err)
+		}
+	}()
+	g.Go(func() error {
+		return udpSrv.ListenAndServe()
+	})
```

Comment:
Can we get away with just calling the method directly?
```suggestion
	g.Go(udpSrv.ListenAndServe)
```